src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/ConnectionSetupSubForms/PunchoutCatalogConnectionSetupForm.php

```diff
@@ -0,0 +1,352 @@
+
+            if (!$idParentCompanyBusinessUnit) {
+                return;
+            }
+
+            $this->updateCompanyUserFieldChoices($form, $idParentCompanyBusinessUnit);
+        };
+
+        $builder->addEventListener(FormEvents::PRE_SUBMIT, $formModificationCallback);
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, $formModificationCallback);
+
+        return $this;
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormInterface $form
+     *
+     * @return void
+     */
+    protected function disableCompanyBusinessUnitField(FormInterface $form): void
+    {
+        $form->add(PunchoutCatalogConnectionSetupTransfer::FK_COMPANY_BUSINESS_UNIT, SelectType::class, array_merge(
+            $this->getCompanyBusinessUnitFieldOptions(),
+            [
+                'mapped' => false,
+            ]
+        ));
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormInterface $form
+     *
+     * @return void
+     */
+    protected function disableCompanyUserField(FormInterface $form): void
+    {
+        $form->add(PunchoutCatalogConnectionSetupTransfer::FK_COMPANY_USER, SelectType::class, array_merge(
+            $this->getCompanyUserFieldOptions(),
+            [
+                'mapped' => false,
+            ]
+        ));
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormInterface $form
+     * @param int $idParentCompanyBusinessUnit
+     *
+     * @return void
+     */
+    protected function updateCompanyBusinessUnitFieldChoices(FormInterface $form, int $idParentCompanyBusinessUnit): void
+    {
+        $form->add(PunchoutCatalogConnectionSetupTransfer::FK_COMPANY_BUSINESS_UNIT, SelectType::class, array_merge(
+            $this->getCompanyBusinessUnitFieldOptions(),
+            [
+                'choices' => $this->getCompanyBusinessUnitChoices($idParentCompanyBusinessUnit),
+            ]
+        ));
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormInterface $form
+     * @param int $idParentCompanyBusinessUnit
+     *
+     * @return void
+     */
+    protected function updateCompanyUserFieldChoices(FormInterface $form, int $idParentCompanyBusinessUnit): void
+    {
+        $form->add(PunchoutCatalogConnectionSetupTransfer::FK_COMPANY_USER, SelectType::class, array_merge(
+            $this->getCompanyUserFieldOptions(),
+            [
+                'choices' => $this->getCompanyUserChoices($idParentCompanyBusinessUnit),
+            ]
+        ));
+    }
+
+    /**
+     * @param int $idParentCompanyBusinessUnit
+     *
+     * @return array
+     */
+    protected function getCompanyBusinessUnitChoices(int $idParentCompanyBusinessUnit): array
+    {
+        $companyBusinessUnitCollectionTransfer = $this->getFactory()
+            ->getCompanyBusinessUnitFacade()
+            ->getCompanyBusinessUnitCollection(
+                (new CompanyBusinessUnitCriteriaFilterTransfer())->setIdParentCompanyBusinessUnit($idParentCompanyBusinessUnit)
```

Comment:
Please move `->setIdParentCompanyBusinessUnit($idParentCompanyBusinessUnit)` to the next line.